iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/ShortBackupRetentionCheck.java

```diff
@@ -138,7 +139,18 @@ private static Double computeRetentionInDays(ContextualObject retentionPolicyObj
   }
 
   private static Predicate<Expression> isNumericValue(DoublePredicate predicate) {
-    return expr -> expr.is(ArmTree.Kind.NUMERIC_LITERAL) && predicate.test(((NumericLiteral) expr).asDouble());
+    return expr -> {
+      if (expr.is(ArmTree.Kind.NUMERIC_LITERAL)) {
+        return predicate.test(((NumericLiteral) expr).asDouble());
+      } else if (expr.is(ArmTree.Kind.UNARY_EXPRESSION)) {
+        UnaryExpression unaryExpression = (UnaryExpression) expr;
+        if (unaryExpression.expression().is(ArmTree.Kind.NUMERIC_LITERAL)) {
+          double factor = unaryExpression.operator().value().equals("-") ? -1 : 1;
+          return predicate.test(((NumericLiteral) unaryExpression.expression()).asDouble() * factor);
```

Comment:
Maybe it could have been a method somewhere in tree utils like `asNumericValueOrNull(UnaryExpression expression)` that would perform kind check and extract the factor?